src/main/java/rx/plugins/RxJavaPlugins.java

```diff
@@ -149,7 +164,27 @@ private static Object getPluginImplementationViaProperty(Class<?> pluginClass) {
          * This will only happen during system startup thus it's okay to use the synchronized
          * System.getProperties as it will never get called in normal operations.
          */
-        String implementingClass = System.getProperty("rxjava.plugin." + classSimpleName + ".implementation");
+        
+        String defaultKey = "rxjava.plugin." + classSimpleName + ".implementation";
+        String implementingClass = props.getProperty(defaultKey);
+        
+        for (Map.Entry<Object, Object> e : props.entrySet()) {
+            String key = e.getKey().toString();
+            if (key.startsWith("rxjava.plugin.") && key.endsWith(".class")) {
+                String value = e.getValue().toString();
+                
+                if (classSimpleName.equals(value)) {
+                    String index = key.substring(0, key.length() - 6).substring(14);
+                    
+                    String implKey = "rxjava.plugin." + index + ".impl";
+                    
+                    implementingClass = props.getProperty(implKey);
```

Comment:
What if user had declared the key, but hadn't declared the value or made an error in the value property name? We can be more restrictive and throw an exception here if key property was declared but value wasn't found. It'll allow detect and prevent errors earlier!